src/Json.Schema.ToDotNet/ClassOrInterfaceGenerator.cs

```diff
@@ -103,7 +103,7 @@ private PropertyDeclarationSyntax CreatePropertyDeclaration(string propertyName)
                 .AddModifiers(GeneratePropertyModifiers(propertyName))
                 .AddAccessorListAccessors(GeneratePropertyAccessors());
 
-            AttributeSyntax[] attributes = GeneratePropertyAttributes(propertyName, info.SerializedName, info.IsRequired);
+            AttributeSyntax[] attributes = GeneratePropertyAttributes(propertyName, info.SerializedName, info.IsRequired, info.DefaultValue);
```

Comment:
>AttributeSyntax[] [](start = 12, length = 17)

I will stop remarking on the point, but the pervasive use of arrays in this code base result in unnecessary memory allocations.

Reply:
Yes, as I mentioned in response to your other comment, many Roslyn APIs take a `params Thing[]`. I'll see if there is another set of Roslyn APIs that use a better pattern.

---
In reply to: [244417551](https://github.com/Microsoft/jschema/pull/60#discussion_r244417551) [](ancestors = 244417551)